web/static/js/actions/channels.js

```diff
@@ -26,31 +26,42 @@ export const receiveChannels = (response) => ({
   response
 })
 
-export const createNuntiumChannel = () => dispatch => {
-  return Promise.all([
-    authorizeWithGuisso(dispatch, 'nuntium', config.nuntium),
-    pigeon.loadPigeonScript(config.nuntium.baseUrl)
-  ])
-    .then(([token, _]) => pigeon.addChannel(token.access_token))
-    .then(nuntiumChannel => {
-      if (nuntiumChannel === null) {
-        return Promise.reject('User cancelled')
-      }
-      dispatch(createChannel({
-        name: nuntiumChannel.name,
-        type: 'sms',
-        provider: 'nuntium',
-        settings: {
-          nuntiumChannel: nuntiumChannel.name
+export const createNuntiumChannel = (() => {
+  let creatingChannel = false
+  return () => dispatch => {
+    if (creatingChannel) {
```

Comment:
- If the check is performed right before `addChannel` multiple pop ups produced because the pop up is opened before that.
- Using a fixed name for the window would avoid opening multiple login windows but it wouldn't prevent the emergence of multiple Add channel pop ups with overlay.